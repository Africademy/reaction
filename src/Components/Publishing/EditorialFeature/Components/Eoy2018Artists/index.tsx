import { color, Serif } from "@artsy/palette"
import { unica } from "Assets/Fonts"
import { ArticleProps } from "Components/Publishing/Layouts/FeatureLayout"
import { Nav } from "Components/Publishing/Nav/Nav"
import { ImageSetPreview } from "Components/Publishing/Sections/ImageSetPreview"
import React from "react"
import styled from "styled-components"
import { Eoy2018ArticleHeader } from "./ArticleHeader"

export const BORDER_WIDTH = 4

export class Eoy2018Artists extends React.Component<ArticleProps> {
  sectionIsHeader = section => {
    const isText = section.type === "text"
    const isHeader = isText && section.body.includes("<h1>")
    return isHeader
  }

  sectionIsHeaderImage = (section, i) => {
    const {
      article: { sections },
    } = this.props
    const isImage = section.type === "image_collection"
    const sectionBefore = i !== 0 && sections[i - 1]
    const sectionBeforeIsHeader = this.sectionIsHeader(sectionBefore)

    return isImage && sectionBeforeIsHeader
  }

  sectionHeader = (section, i) => {
    const {
      article: { sections },
    } = this.props
    const sectionAfter = sections[i + 1]
    const sectionAfterisImage = this.sectionIsHeaderImage(sectionAfter, i + 1)
    const src = sectionAfterisImage && sectionAfter.images[0].url

    return (
      <ArtistHeaderSection key={i}>
        <ArtistHeaderTitle dangerouslySetInnerHTML={{ __html: section.body }} />
        <ArtistHeaderImg src={src} />
      </ArtistHeaderSection>
    )
  }

  sectionText = (section, i) => {
    return (
      <TextSection size="5" key={i}>
        <div dangerouslySetInnerHTML={{ __html: section.body }} />
      </TextSection>
    )
  }

  sectionImageSet = (section, i) => {
    return (
      <ImageSetWrapper key={i}>
        <ImageSetPreview section={section} />
      </ImageSetWrapper>
    )
  }

  getSections = () => {
    const {
      article: { sections },
    } = this.props

    const renderedSections = sections.map((section, i) => {
      const isHeader = this.sectionIsHeader(section)
      const isHeaderImage = this.sectionIsHeaderImage(section, i)

      if (isHeader) {
        return this.sectionHeader(section, i)
      } else if (isHeaderImage) {
        return null
      } else {
        if (section.type === "text") {
          return this.sectionText(section, i)
        } else if (section.type === "image_set") {
          return this.sectionImageSet(section, i)
        } else {
          return null
        }
      }
    })
    return renderedSections
  }

  render() {
    return (
      <React.Fragment>
        <Nav canFix />

        <ArticleWrapper>
          <Eoy2018ArticleHeader />

          <ArticleContent>{this.getSections()}</ArticleContent>
        </ArticleWrapper>
      </React.Fragment>
    )
  }
}

const ArticleWrapper = styled.div`
  padding: 0 40px;
`

const ArticleContent = styled.div`
  padding: 40px 0;
  border-left: ${BORDER_WIDTH}px solid ${color("purple100")};

  blockquote {
    ${unica("s34")};
  }
`

const ArtistHeaderSection = styled.div`
  height: 55vh;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;
  display: flex;
  margin-bottom: 40px;
`

const ArtistHeaderTitle = styled.div`
  flex: 1;
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};

  h1 {
    ${unica("s65")};
    width: 100%;
    height: 50%;
    padding: 20px;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
  }

  h2 {
    ${unica("s25")};
    width: 50%;
    height: 50%;
    display: inline-flex;
    padding: 20px;
    &:last-child {
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    }
    &:hover {
      background: ${color("purple100")};
      color: white;
    }
  }
`

const ArtistHeaderImg = styled.div<{ src?: string }>`
  flex: 1;
  ${props =>
    props.src &&
    `
    background: url(${props.src});
    background-size: cover;
    background-position: center;
  `};
`

const TextSection = styled(Serif)`
  max-width: 75%;
  margin-left: auto;
  padding-bottom: 60px;

  p {
    font-size: 24;
  }
`
const ImageSetWrapper = styled.div`
  margin-bottom: 60px;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;
`
