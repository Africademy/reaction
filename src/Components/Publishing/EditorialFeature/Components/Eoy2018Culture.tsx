import { Box, Flex, Sans } from "@artsy/palette"
import { unica } from "Assets/Fonts"
import { Author, StyledAuthor } from "Components/Publishing/Byline/Author"
import { Date, DateContainer } from "Components/Publishing/Byline/Date"
import { Share } from "Components/Publishing/Byline/Share"
import { getArticleFullHref } from "Components/Publishing/Constants"
import {
  ArticleProps,
  FeatureLayout,
} from "Components/Publishing/Layouts/FeatureLayout"
import { Nav, NavContainer } from "Components/Publishing/Nav/Nav"
import React from "react"
import styled from "styled-components"

export const Eoy2018Culture: React.SFC<ArticleProps> = props => {
  const { article } = props
  return (
    <ArticleWrapper>
      <Nav canFix color="black" backgroundColor="white" />

      <ArticleHeader>
        <ArticleTitle size="16" weight="medium">
          <div>Year In Visual Culture</div>
          <div>2018</div>
        </ArticleTitle>
        <HeaderImg src={article.hero_section.url} />
      </ArticleHeader>

      <Intro>
        <Box width={"20%"}>
          <Share
            url={getArticleFullHref(article.slug)}
            title={article.social_title || article.thumbnail_title}
          />
        </Box>
        <Flex>
          <Author authors={article.authors} />
          <Date date={article.published_at} />
        </Flex>
      </Intro>

      <Box>
        <div dangerouslySetInnerHTML={{ __html: article.sections[0].body }} />
      </Box>
      <FeatureLayout {...props} />
    </ArticleWrapper>
  )
}

const ArticleWrapper = styled.div`
  ${NavContainer} {
    border-bottom-width: 6px;
    border-top: 6px solid black;
  }

  blockquote {
    ${unica("s34")};
  }
`

const ArticleTitle = styled(Sans)`
  text-transform: uppercase;
  padding: 20px 40px;
  border-bottom: 6px solid;
`

const Intro = styled(Flex)`
  align-items: center;
  padding: 20px 40px;

  ${DateContainer} {
    margin-top: 0;
  }
  ${StyledAuthor} {
    margin-top: 0;
  }
`

const HeaderImg = styled.img`
  height: 100%;
  object-fit: cover;
  object-position: center;
  width: 100%;
`

const ArticleHeader = styled.div`
  height: calc(100vh - 46px);
  overflow: hidden;
  border-bottom: 6px solid black;
`
