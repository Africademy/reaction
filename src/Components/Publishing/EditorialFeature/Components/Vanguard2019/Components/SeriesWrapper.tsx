import { Box, Sans } from "@artsy/palette"
import { ArticleData } from "Components/Publishing/Typings"
import React from "react"
import { VanguardArtistWrapper } from "./ArtistWrapper"

export const VanguardSeriesWrapper: React.SFC<{
  article: ArticleData
}> = props => {
  const { relatedArticles, title } = props.article
  return (
    <Box>
      <Sans size="6">Sub-Series Article: {title}</Sans>
      {/** map sub-series artist articles */}
      {relatedArticles &&
        relatedArticles.map(artistArticle => (
          <VanguardArtistWrapper article={artistArticle} />
        ))}
    </Box>
  )
}
