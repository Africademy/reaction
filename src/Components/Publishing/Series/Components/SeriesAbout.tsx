import React, { Component  } from "react"
import styled, { StyledFunction } from "styled-components"
import { pMedia } from "../../../Helpers"
import { Fonts } from "../../Fonts"
import { PartnerBlock } from '../../Partner/PartnerBlock'
import { Text } from '../../Sections/Text'

interface Props {
  color?: string,
  series?: any
}

export class SeriesAbout extends Component<Props, null> {
  public static defaultProps: Partial<Props>

  render () {
    const { series } = this.props
    const { about, sponsor } = series

    return (
      <SeriesAboutContainer className='SeriesAbout'>

          <div className="col col--first">
            <Title>About the Series</Title>
            {sponsor &&
              <PartnerBlock
                logo={sponsor.partner_logo}
                url={sponsor.partner_logo_link}
                trackingData={{
                  type: 'external link',
                  destination_path: sponsor.partner_logo_link
                }}
              />
            }
          </div>

          <div className="col col--last">
            <Text layout='standard' html={about} />
            {sponsor &&
              <PartnerBlock
                logo={sponsor.partner_logo}
                url={sponsor.partner_logo_link}
                trackingData={{
                  type: 'external link',
                  destination_path: sponsor.partner_logo_link
                }}
              />
            }
          </div>

      </SeriesAboutContainer>
    )
  }
}

SeriesAbout.defaultProps = {
  color: 'black'
}

const Div: StyledFunction<Props & React.HTMLProps<HTMLDivElement>> = styled.div

const SeriesAboutContainer = Div`
  color: ${props => props.color};
  display: flex;
  justify-content: space-between;

  .col {
    padding: 0;
    &--first {
      width: 30%;
      display: flex;
      justify-content: space-between;
      flex-direction: column;

      .PartnerBlock {
        display: block;
      }
    }
    &--last {
      width: 60%;
      .PartnerBlock {
        display: none;
      }
    }
  }

  .article__text-section {
    min-width: 100%;
  }

  ${props => pMedia.md`
    display: block;

    .col {
      width: 100%;
      &--first .PartnerBlock {
        display: none;
      }
      &--last .PartnerBlock {
        display: block;
        margin-top: 60px;
      }
    }
  `}
`

const Title = styled.div`
  ${Fonts.unica("s32", "medium")}
  ${props => pMedia.md`
    margin-bottom: 20px;
  `}
`
