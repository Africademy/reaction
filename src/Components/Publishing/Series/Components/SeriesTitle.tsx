import React, { Component  } from "react"
import styled, { StyledFunction } from "styled-components"
import { pMedia } from "../../../Helpers"
import { Fonts } from "../../Fonts"
import { PartnerBlock } from '../../Partner/PartnerBlock'

interface Props {
  color?: string,
  series?: any
}

export class SeriesTitle extends Component<Props, null> {
  public static defaultProps: Partial<Props>

  render () {
    const { color, series } = this.props
    const { sponsor } = series

    return (
      <SeriesTitleContainer className='SeriesTitle' color={color}>

        <Title>{series.title}</Title>

        {sponsor &&
          <PartnerBlock
            logo={sponsor.partner_logo}
            url={sponsor.partner_logo_link}
            trackingData={{
              type: 'external link',
              destination_path: sponsor.partner_logo_link
            }}
          />
        }

      </SeriesTitleContainer>
    )
  }
}

SeriesTitle.defaultProps = {
  color: 'black'
}

const Div: StyledFunction<Props & React.HTMLProps<HTMLDivElement>> = styled.div

const SeriesTitleContainer = Div`
  color: ${props => props.color};
  text-align: center;

  .PartnerBlock img {
    max-width: 170px;
    padding-top: 5px;
  }

  ${props => pMedia.md`
    .PartnerBlock img {
      max-width: 200px;
    }
  `}
`

const Title = styled.div`
  ${Fonts.unica("s120")}
  margin-bottom: 30px;

  ${props => pMedia.md`
    ${Fonts.unica("s65")}
    margin-bottom: 20px;
  `}
`
