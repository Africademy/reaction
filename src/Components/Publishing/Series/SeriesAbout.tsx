import React, { Component } from "react"
import { Col } from "react-styled-flexboxgrid"
import styled, { StyledFunction } from "styled-components"
import { media } from "../../Helpers"
import { Fonts } from "../Fonts"
import { PartnerBlock, PartnerBlockContainer } from '../Partner/PartnerBlock'
import { Text } from '../Sections/Text'

interface Props {
  article?: any
  color?: string
  editDescription?: any
}

export class SeriesAbout extends Component<Props, null> {
  public static defaultProps: Partial<Props>

  render() {
    const { article, color, editDescription } = this.props
    const { series_description, sponsor } = article

    return (
      <SeriesAboutContainer color={color}>
        <StyledCol xs={12} sm={4}>
          <Title>About the Series</Title>
          {sponsor &&
            <PartnerBlock
              logo={sponsor.partner_logo}
              url={sponsor.partner_logo_link}
              trackingData={{
                type: 'external link',
                destination_path: sponsor.partner_logo_link
              }}
            />
          }
        </StyledCol>
        <StyledCol xs={12} sm={8}>
          {editDescription
            ? <Text layout='standard'>{editDescription}</Text>
            : <Text layout='standard' html={series_description} />
          }
          {sponsor &&
            <PartnerBlock
              logo={sponsor.partner_logo}
              url={sponsor.partner_logo_link}
              trackingData={{
                type: 'external link',
                destination_path: sponsor.partner_logo_link
              }}
            />
          }
        </StyledCol>
      </SeriesAboutContainer>
    )
  }
}

SeriesAbout.defaultProps = {
  color: 'black'
}

interface ColProps {
  first?: boolean
}

const Div: StyledFunction<Props & ColProps & React.HTMLProps<HTMLDivElement>> = styled.div
export const SeriesAboutContainer = Div`
  color: ${props => props.color};
  max-width: 1200px;
  display: flex;
`
const StyledCol = styled(Col) `
  flex: 1;
  ${PartnerBlockContainer} {
    display: none;
  }

  &:first-of-type {
    display: flex;
    justify-content: space-between;
    flex-direction: column;
    ${PartnerBlockContainer} {
      display: block;
    }
  }

  ${props => media.sm`
    &:first-of-type {
      ${PartnerBlockContainer} {
        display: none;
      }
    }

    ${PartnerBlockContainer} {
      margin-top: 60px;
      display: block;
    }
  `}
`

const Title = styled.div`
  ${Fonts.unica("s32", "medium")}
  ${props => media.sm`
    margin-bottom: 20px;
  `}
`
