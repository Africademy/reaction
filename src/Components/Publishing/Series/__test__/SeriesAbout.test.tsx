import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { SeriesArticle, SeriesArticleSponsored } from "../../Fixtures/Articles"
import { SeriesAbout } from "../Components/SeriesAbout"

it("renders a series about properly", () => {
  const component = renderer.create(<SeriesAbout series={SeriesArticle} />).toJSON()
  expect(component).toMatchSnapshot()
})

it("renders a sponsored series about properly", () => {
  const component = renderer.create(<SeriesAbout series={SeriesArticleSponsored} />).toJSON()
  expect(component).toMatchSnapshot()
})

it("Renders partner block for a sponsored series", () => {
  const component = mount(<SeriesAbout series={SeriesArticleSponsored} />)
  expect(component.find('.PartnerBlock').length).not.toBe(0)
})

it("Does not render partner block for an unsponsored series", () => {
  const component = mount(<SeriesAbout series={SeriesArticle} />)
  expect(component.find('.PartnerBlock').length).toBe(0)
})
