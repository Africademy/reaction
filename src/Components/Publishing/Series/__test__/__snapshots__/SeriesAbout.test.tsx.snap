// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a series about properly 1`] = `
.c4 {
  position: relative;
  width: 100%;
}

.c4 a {
  color: black;
  text-decoration: none;
  position: relative;
  background-image: linear-gradient(to bottom,transparent 0,#333 1px,transparent 0);
  background-size: 1.25px 4px;
  background-repeat: repeat-x;
  background-position: bottom;
}

.c4 a:hover {
  color: #999;
}

.c4 p,
.c4 ul,
.c4 ol {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
  padding-top: 1em;
  padding-bottom: 1em;
  margin: 0;
  font-style: inherit;
}

.c4 p:first-child {
  padding-top: 0;
}

.c4 p:last-child {
  padding-bottom: 0;
}

.c4 ul,
.c4 ol {
  padding-left: 1em;
}

.c4 li {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
  padding-top: .5em;
  padding-bottom: .5em;
}

.c4 h1 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 40px;
  line-height: 1.1em;
  font-weight: normal;
  padding-top: 107px;
  padding-bottom: 46px;
  margin: 0;
  position: relative;
  text-align: center;
}

.c4 h1:before {
  content: "";
  width: 15px;
  height: 15px;
  background: black;
  border-radius: 50%;
  position: absolute;
  top: 69px;
  right: calc(50% - 7.5px);
}

.c4 h2 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 32px;
  line-height: 1.1em;
  font-weight: normal;
  margin: 0;
}

.c4 h2 a {
  background-size: 1.25px 1px;
}

.c4 h3 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
  font-weight: normal;
  padding-top: 23px;
  margin: 0;
}

.c4 h3 strong {
  font-weight: normal;
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
}

.c4 h3 a {
  background-size: 1.25px 1px;
}

.c4 blockquote {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 40px;
  line-height: 1.1em;
  text-align: left;
  font-weight: normal;
  padding-top: 46px;
  padding-bottom: 46px;
  margin: 0;
  word-break: break-word;
}

.c4 .content-end {
  display: inline-block;
  content: "";
  width: 12px;
  height: 12px;
  background: black;
  border-radius: 50%;
  margin-left: 12px;
}

.c4 .artist-follow {
  vertical-align: middle;
  margin-left: 10px;
  cursor: pointer;
  background: none transparent;
}

.c4 .artist-follow:before {
  font-family: "artsy-icons";
  content: "î˜©";
  vertical-align: -8.5px;
  line-height: 32px;
  font-size: 32px;
}

.c4 .artist-follow:after {
  content: "Follow";
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
  text-transform: none;
}

.c0 {
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  width: 30%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  width: 60%;
}

.c3 .file__PartnerBlockContainer-s8dqalt-0 {
  display: none;
}

.c2 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 32px;
  line-height: 1.1em;
}

@media (max-width:600px) {
  .c4 p,
  .c4 ul,
  .c4 ol {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
  }

  .c4 li {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
  }

  .c4 h1 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 34px;
    line-height: 1.1em;
  }

  .c4 h2 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }

  .c4 h3 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
    line-height: 1.5em;
  }

  .c4 h3 strong {
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }

  .c4 blockquote {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 34px;
    line-height: 1.1em;
  }

  .c4 .content-start {
    font-size: 55px;
  }
}

@media (max-width:900px) {
  .c0 {
    display: block;
  }
}

@media (max-width:900px) {
  .c1 {
    width: 100%;
  }

  .c1 .file__PartnerBlockContainer-s8dqalt-0 {
    margin-top: 60px;
    display: none;
  }
}

@media (max-width:900px) {
  .c3 {
    width: 100%;
  }

  .c3 .file__PartnerBlockContainer-s8dqalt-0 {
    margin-top: 60px;
    display: block;
  }
}

@media (max-width:900px) {
  .c2 {
    margin-bottom: 20px;
  }
}

<div
  className="SeriesAbout c0"
  color="black"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      About the Series
    </div>
  </div>
  <div
    className="c3"
  >
    <div
      className="article__text-section c4"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Integer posuere erat a ante venenatis dapibus posuere velit aliquet. Curabitur blandit tempus porttitor. Donec ullamcorper nulla non metus auctor fringilla. Donec ullamcorper nulla non metus auctor fringilla. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Maecenas faucibus mollis interdum. Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p><p>Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Curabitur blandit tempus porttitor. Sed posuere consectetur est at lobortis. Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Nullam quis risus eget urna mollis ornare vel eu leo.</p><p>Donec sed odio dui. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Nullam quis risus eget urna mollis ornare vel eu leo. Nulla vitae elit libero, a pharetra augue.</p>",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`renders a sponsored series about properly 1`] = `
.c4 img {
  max-width: 220px;
  padding-top: 20px;
}

.c5 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c7 {
  position: relative;
  width: 100%;
}

.c7 a {
  color: black;
  text-decoration: none;
  position: relative;
  background-image: linear-gradient(to bottom,transparent 0,#333 1px,transparent 0);
  background-size: 1.25px 4px;
  background-repeat: repeat-x;
  background-position: bottom;
}

.c7 a:hover {
  color: #999;
}

.c7 p,
.c7 ul,
.c7 ol {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
  padding-top: 1em;
  padding-bottom: 1em;
  margin: 0;
  font-style: inherit;
}

.c7 p:first-child {
  padding-top: 0;
}

.c7 p:last-child {
  padding-bottom: 0;
}

.c7 ul,
.c7 ol {
  padding-left: 1em;
}

.c7 li {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
  padding-top: .5em;
  padding-bottom: .5em;
}

.c7 h1 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 40px;
  line-height: 1.1em;
  font-weight: normal;
  padding-top: 107px;
  padding-bottom: 46px;
  margin: 0;
  position: relative;
  text-align: center;
}

.c7 h1:before {
  content: "";
  width: 15px;
  height: 15px;
  background: black;
  border-radius: 50%;
  position: absolute;
  top: 69px;
  right: calc(50% - 7.5px);
}

.c7 h2 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 32px;
  line-height: 1.1em;
  font-weight: normal;
  margin: 0;
}

.c7 h2 a {
  background-size: 1.25px 1px;
}

.c7 h3 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
  font-weight: normal;
  padding-top: 23px;
  margin: 0;
}

.c7 h3 strong {
  font-weight: normal;
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
}

.c7 h3 a {
  background-size: 1.25px 1px;
}

.c7 blockquote {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 40px;
  line-height: 1.1em;
  text-align: left;
  font-weight: normal;
  padding-top: 46px;
  padding-bottom: 46px;
  margin: 0;
  word-break: break-word;
}

.c7 .content-end {
  display: inline-block;
  content: "";
  width: 12px;
  height: 12px;
  background: black;
  border-radius: 50%;
  margin-left: 12px;
}

.c7 .artist-follow {
  vertical-align: middle;
  margin-left: 10px;
  cursor: pointer;
  background: none transparent;
}

.c7 .artist-follow:before {
  font-family: "artsy-icons";
  content: "î˜©";
  vertical-align: -8.5px;
  line-height: 32px;
  font-size: 32px;
}

.c7 .artist-follow:after {
  content: "Follow";
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
  text-transform: none;
}

.c0 {
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  width: 30%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  width: 60%;
}

.c6 .c3 {
  display: none;
}

.c2 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 32px;
  line-height: 1.1em;
}

@media (max-width:720px) {
  .c4 img {
    max-width: 195px;
  }
}

@media (max-width:720px) {
  .c5 {
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
  }
}

@media (max-width:600px) {
  .c7 p,
  .c7 ul,
  .c7 ol {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
  }

  .c7 li {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
  }

  .c7 h1 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 34px;
    line-height: 1.1em;
  }

  .c7 h2 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }

  .c7 h3 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
    line-height: 1.5em;
  }

  .c7 h3 strong {
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }

  .c7 blockquote {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 34px;
    line-height: 1.1em;
  }

  .c7 .content-start {
    font-size: 55px;
  }
}

@media (max-width:900px) {
  .c0 {
    display: block;
  }
}

@media (max-width:900px) {
  .c1 {
    width: 100%;
  }

  .c1 .c3 {
    margin-top: 60px;
    display: none;
  }
}

@media (max-width:900px) {
  .c6 {
    width: 100%;
  }

  .c6 .c3 {
    margin-top: 60px;
    display: block;
  }
}

@media (max-width:900px) {
  .c2 {
    margin-bottom: 20px;
  }
}

<div
  className="SeriesAbout c0"
  color="black"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      About the Series
    </div>
    <div
      className="PartnerBlock c3 c4"
    >
      <div
        className="c5"
      >
        Presented In Partnership With
      </div>
      <a
        href="http://artsy.net"
        onClick={[Function]}
        target="_blank"
      >
        <img
          src="https://artsy-media-uploads.s3.amazonaws.com/eZqsrnodcIlyyJzmRqLm4A%2FBombay_Sapphire_logo.png"
        />
      </a>
    </div>
  </div>
  <div
    className="c6"
  >
    <div
      className="article__text-section c7"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Integer posuere erat a ante venenatis dapibus posuere velit aliquet. Curabitur blandit tempus porttitor. Donec ullamcorper nulla non metus auctor fringilla. Donec ullamcorper nulla non metus auctor fringilla. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Maecenas faucibus mollis interdum. Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p><p>Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Curabitur blandit tempus porttitor. Sed posuere consectetur est at lobortis. Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Nullam quis risus eget urna mollis ornare vel eu leo.</p><p>Donec sed odio dui. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Nullam quis risus eget urna mollis ornare vel eu leo. Nulla vitae elit libero, a pharetra augue.</p>",
          }
        }
      />
    </div>
    <div
      className="PartnerBlock c3 c4"
    >
      <div
        className="c5"
      >
        Presented In Partnership With
      </div>
      <a
        href="http://artsy.net"
        onClick={[Function]}
        target="_blank"
      >
        <img
          src="https://artsy-media-uploads.s3.amazonaws.com/eZqsrnodcIlyyJzmRqLm4A%2FBombay_Sapphire_logo.png"
        />
      </a>
    </div>
  </div>
</div>
`;
